src/components/VolumeChart.tsx

```diff
@@ -0,0 +1,217 @@
+import { useEffect, useRef, useState } from "react";
+import Chart from "chart.js/auto";
+import "chartjs-adapter-date-fns";
+import {
+  subDays,
+  startOfDay,
+  format,
+  startOfToday,
+  subMonths,
+} from "date-fns";
+import { cn } from "@/lib/utils";
+import { useTheme } from "next-themes";
+import { Skeleton } from "./ui/skeleton";
+import { Currency } from "@/lib/types";
+
+interface VolumeChartProps {
+  totalVolumes: number[][];
+  timeframe: number;
+  currency: [Currency, number];
+}
+
+const VolumeChart = ({
+  totalVolumes,
+  timeframe,
+  currency,
+}: VolumeChartProps) => {
+  const { theme } = useTheme();
+  const chartRef = useRef<HTMLCanvasElement>(null);
+  const chartInstanceRef = useRef<Chart | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    setIsLoading(true);
+    const timeframes = ["1D", "7D", "14D", "1M"];
+    if (chartInstanceRef.current) {
+      chartInstanceRef.current.destroy();
+      chartInstanceRef.current = null;
+    }
+
+    if (chartRef.current) {
+      const chartContext = chartRef.current.getContext("2d");
+      let startDate;
+      switch (timeframes[timeframe]) {
+        case "1D":
+          startDate = subDays(startOfToday(), 1);
+          break;
+        case "7D":
+          startDate = subDays(startOfToday(), 7);
+          break;
+        case "14D":
+          startDate = subDays(startOfToday(), 14);
+          break;
+        case "1M":
+          startDate = subMonths(startOfToday(), 1);
+          break;
+        default:
+          startDate = subMonths(startOfToday(), 1);
```

Comment:
same here, you could extract this whole logic in a function

Reply:
function created in @/lib/utils.ts and implemented in both files